ptvsd/wrapper.py

```diff
@@ -613,7 +613,21 @@ def __init__(self, socket, pydevd, logfile=None, killonclose=True):
         self.event_loop_thread.daemon = True
         self.event_loop_thread.start()
 
+    def __input(self, message):
```

Comment:
To facilitate testing this, I'd recommend making the "input" function an attribute of VSCodeMessageProcessor:

    def __init__(self, ..., inputfunc=None, ...):
        ...
        try:
            self.inputfunc = raw_input
        except NameError:
            self.inputfunc = input
        ...

Then in _handle_exit() just call self.inputfunc().

Then for the test machinery you'll need to pass an appropriate inputfunc.  Look for any place we pass `killonclose=False`.  There should be a couple of spots, both under tests/helpers/pydevd/.

Also, FYI, using "__"-prefixed names is discouraged in Python.  A single underscore should be sufficient.